automated install/basic-install.sh

```diff
@@ -1246,8 +1246,8 @@ setAdminFlag() {
             ;;
     esac
 
-    # If the user wants to install the Web admin interface (i.e. it has not been deselected above)
-    if [[ "${INSTALL_WEB_INTERFACE}" == true ]]; then
+    # If the user wants to install the Web admin interface (i.e. it has not been deselected above) and did not deselect the web server via command-line argument
+    if [[ "${INSTALL_WEB_SERVER}" == true ]]; then
```

Comment:
I think it would be better to check both `WEB_SERVER` and `WEB_INTERFACE` here.

It is odd to check for `WEB_SERVER=true` on line 1250 and then set `WEB_SERVER` to `true` on line 1270. That says: "You've already said you want the web server, do you really want the web server?"

I can also see the case of wanting the web interface repo and not wanting `lighttpd`. I don't see the case of not wanting the web interface repo and still wanting `lighttpd`, at least that's not a situation we should be handling. 

Cases:
 
| WEB INTERFACE | WEB SERVER | SUPPORTED|
| --- | --- | --- |
| :white_check_mark: | :white_check_mark: | :white_check_mark: |
| :white_check_mark: | :x: | :white_check_mark: |
| :x: | :white_check_mark: | :x: |

Reply:
> That says: "You've already said you want the web server, do you really want the web server?"

`INSTALL_WEB_SERVER=true` is set as hardcoded default, so it basically means "not selected yet". But I agree it would be better understandable to have it unset or something by default, so that `true` really means "true was selected" and `false` likewise.

> I can also see the case of wanting the web interface repo and not wanting lighttpd. I don't see the case of not wanting the web interface repo and still wanting lighttpd, at least that's not a situation we should be handling.

If the web server is deselected, `INSTALL_WEB_SERVER=false` is set as well, so that case is covered as well, the same way it was before. So I think the needed cases you listed above are supported with this PR, I mean the logic now is again exactly as it was before the switch to `dialog`.

Reply:
> I mean the logic now is again exactly as it was before the switch to `dialog`.

True but that doesn't mean the logic can't be improved.

Reply:
I agree with Dan. To make the code more understandable, we should check for both.

```suggestion
    # If the user wants to install the Web admin interface (i.e. it has not been deselected above) and did not deselect the web server via command-line argument
    if [[ "${INSTALL_WEB_INTERFACE}" == true ]] && [[ "${INSTALL_WEB_SERVER}" == true ]] ; then
```

Reply:
Done. While being redundant at the moment, I agree it is better understandable, fits better to the code commend above and will better stand changes, if the logic gets changed.